File: message_queue_sender.py
import sysv_ipc
key = sysv_ipc.ftok("msgqfile", 65)
queue = sysv_ipc.MessageQueue(key, sysv_ipc.IPC_CREAT | 0o666)
message = "Hello from the background sender!"
queue.send(message)
print(f"Message sent: {message}")

File: message_queue_sender.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/ipc.h>
#include <sys/msg.h>
#include <sys/types.h>
#include <unistd.h>
struct msgbuf {
long mtype;
char mtext[100];
};
int main() {
key_t key;
int msgid;
key = ftok("msgqfile", 65);
msgid = msgget(key, 0666 | IPC_CREAT);
struct msgbuf message;
message.mtype = 1;
strcpy(message.mtext, "Hello from the background sender!");
msgsnd(msgid, &message, sizeof(message), 0);
printf("Message sent: %s\n", message.mtext);
return 0;
}
